### DISCUSSION TITLE
Customize action of existing provider

### DISCUSSION THREAD

**Discussion by circld** [liuchengxu/vim-clap#647] [Q&A] [answered]
Is there a way to customize the action of an existing provider? Concretely, I personally find `bcommits` much more useful when hitting `enter` opens a diff comparing the latest file to the commit in question:

![image](https://user-images.githubusercontent.com/5423134/109438501-ede0a800-79ef-11eb-975e-c76fca242e33.png)

I've made this change in-place, but i recognize that this is not ideal. I can certainly copy the entire provider as a new custom provider, but that seems heavy handed for such a limited change.

Is that my only recourse, or do I have other options here?

**Comment by liuchengxu** [comment, marked as answer]
You can override the existing provider in your vimrc like this:

```vim
  let g:MyBcommits = g:clap#provider#bcommits#

  function! g:MyBcommits.sink(line) abort
    echom 'New Sink'
  endfunction
```

**Comment by circld** [comment]
Awesome, I **knew** there was a better way to accomplish this.
